src/dc-base-component.ts

```diff
@@ -0,0 +1,123 @@
+import {
+    getElementOptions,
+    getElementRefs, ReferencesCollection,
+} from './dc-dom';
+
+/**
+ * @typedef {Object.<string, {(HTMLElement|HTMLElement[]|Object.<string, HTMLElement>)}>} ReferencesCollection
+ */
+
+/**
+ * Interface for classes that can be initialized by factory.
+ * @interface Component
+*/
+/**
+ * Init Component
+ * @method
+ * @name Component#init
+ *
+ */
+/**
+ * Destroy Component
+ * @method
+ * @name Component#destroy
+ *
+ */
+/**
+ * @method
+ * @name Component#getElement
+ * @return {HTMLElement} element
+ */
+/**
+ * @method
+ * @name Component#getNamespace
+ * @static
+ */
+
+
+interface DcListener {
+    elem: HTMLElement;
+    eventName: string;
+    eventCallback: CallableFunction;
+}
+
+/**
+ * Base component.
+ * @class
+ * @implements {DcBaseComponent}
+ */
+
+
+class DcBaseComponent {
```

Comment:
@Ofigelov can we handle the base component class as abstract class with some abstract properties? what do you think?

Reply:
I have tried to do it already, but since we need to have an access to static elements of nested Classes, and if DcBaseComponent is abstract, then TS will throw an error about that we can't create instance of abstract class and use static elements of abstract class. So I don't know how to solve it.